c++/src/kj/common.h

```diff
@@ -1910,22 +1910,24 @@ class ArrayPtr: public DisallowConstCopyIfNotConst<T> {
     return { reinterpret_cast<PropagateConst<T, char>*>(ptr), size_ * sizeof(T) };
   }
 
-  inline bool operator==(decltype(nullptr)) const { return size_ == 0; }
+  inline constexpr bool operator==(decltype(nullptr)) const { return size_ == 0; }
 
-  inline bool operator==(const ArrayPtr& other) const {
+  inline constexpr bool operator==(const ArrayPtr& other) const {
     if (size_ != other.size_) return false;
+#if __has_feature(cxx_constexpr_string_builtins)
```

Comment:
as it is now means msvc wouldn't use memcmp optimization

Reply:
FWIW I'm not as worried about squeezing all the optimization juice out of MSVC builds.